### PR TITLE
Add a role for regularly downloading SSH keys

### PR DESCRIPTION
**Pull request by philwills**

### CODE REVIEW

**Comment by philwills** [comment]
Note that this leaves specifying the bucket to recipe configuration, so isn't dependent on the changes @sihil, @philmcmahon and I discussed yesterday.

File: roles/github-ssh-keys/templates/install-keys.sh.template [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -e
+
+aws s3 cp s3://{{github_keys_bucket}}/{{github_team_name}}/authorized_keys /tmp/authorized_keys
+
+cp /tmp/authorized_keys /home/{{ssh_user}}/.ssh/authorized_keys
+
+chmod 600 /home/{{ssh_user}}/.ssh/authorized_keys
```
Comment (sihil) [inline roles/github-ssh-keys/templates/install-keys.sh.template]: these `cp` and `chmod` commands could be combined into a single `install` command

**Comment by sihil** [comment]
Can there be a README with some details in such as how to check that this is running / has run and any configuration notes?

File: roles/github-ssh-keys/files/update-keys.timer [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Update SSH keys every half hour
+
+[Timer]
+OnBootSec=30 seconds
+OnUnitActiveSec=30 minutes
+
+[Install]
+WantedBy=timers.target
```
Comment (sihil) [inline roles/github-ssh-keys/files/update-keys.timer]: This is really nice. Can very much see this replacing cron completely.

File: roles/github-ssh-keys/templates/install-keys.sh.template (hunk 1 above)
Comment (philwills) [inline roles/github-ssh-keys/templates/install-keys.sh.template]: Ooh, fancy, I'll give it a go.

**Comment by philwills** [comment]
Yea, a README sounds good.

**Comment by philwills** [comment]
Something that occurred to me is that this has nothing to do with Github, so I'm tempted to rename to S3 SSH keys and the parameters:

`github_keys_bucket` -> `ssh_keys_bucket`
`github_team_name` -> `ssh_keys_prefix`

I can see good arguments both ways on this, so would welcome feedback.

**Comment by sihil** [comment]
I would totally advocate for making it generic and simply documenting the expected format of the bucket.

**Comment by philmcmahon** [comment]
Yep, that sounds sensible.

File: roles/s3-ssh-keys/files/update-keys.timer [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Update SSH keys every half hour
+
+[Timer]
+OnStartupSec=1 minute
```
Comment (philwills) [inline roles/s3-ssh-keys/files/update-keys.timer]: I tried making this sooner, but it seems that if you try to do it too quickly the aws instance profile credentials are unavailable.

**Comment by sihil** [comment]
Perfect. 👍

**Comment by philmcmahon** [comment]
Brilliant! 🔐

**Pull request merged**